docs/content/concepts/partitions-schedules-sensors/sensors.mdx

```diff
@@ -334,6 +393,25 @@ def test_my_directory_sensor_cursor():
         assert validate_run_config(log_file_job, run_request.run_config)
 ```
 
+#### Testing sensors with resources
+
+For sensors which utilize [resources](/concepts/resources), you can provide the necessary resources when invoking the sensor function.
+
+Below is a test for the `process_new_users_sensor` that we defined in the [Using resources in sensors](#using-resources-in-sensors) section, which uses the `users_api` resource.
+
+```python file=/concepts/resources/pythonic_resources.py startafter=start_test_resource_on_sensor endbefore=end_test_resource_on_sensor dedent=4
+from dagster import build_sensor_context, validate_run_config
+
+def test_process_new_users_sensor():
+    class MockUsersAPI(ConfigurableResource):
```

Comment:
Hmmm does this work without inheriting from UsersAPI?

Also this is technically a "fake" rather than a "mock".

This inspired me to write up some thoughts in https://github.com/dagster-io/internal/discussions/5447. Not particularly satisfied with the patterns emerging.

Reply:
Responding in thread. I agree that this pattern is pretty rough right now.